### PR TITLE
Update to the java emitter to support new getAllTestResults and abort tests

### PR DESCRIPTION
**Pull request by Artem-applitools**
It also required small changes to the tests so it was possible to generate in java

### CODE REVIEW

File: coverage-tests.js [hunk 1]
```diff
@@ -2029,7 +2029,7 @@ test('should return aborted tests in getAllTestResults', {
     eyes.open({appName: 'Applitools Eyes SDK'})
     eyes.check()
     const abortResult = eyes.abort()
-    assert.equal(abortResult.getIsAborted(), true)
+    assert.equal(abortResult.isAborted, true)
```
Comment (amitzur) [inline coverage-tests.js:2032]: Oh, so in Java it's `.isAborted` and `setAborted`? In JS it's `.getIsAborted` and `.setIsAborted`.

**Review by amitzur — APPROVED** [review]

File: coverage-tests.js (hunk 1 above)
Comment (Artem-applitools) [reply coverage-tests.js:2032]: Yes, but for the generation the biggest problem that it was a function, it requires much more changes to handle it in such case. 
For the `abortResult.isAborted` we need in the emitter describe only type: 
```
abort() {
            return addCommand(java`eyes.abort();`).type({
    type: "TestResults",
    schema: {
        isAborted: "Boolean"
    }
} )
}
```
But for the `abortResult.getIsAborted()`
We need to add method for the returned type

```
abort() {
            return addCommand(java`eyes.abort();`).type({
    type: "TestResults"
} ).methods(getIsAborted: (target) => addCommand(java`${target}.isAborted()`).type({type: "Boolean"}))
}
```
The main reason for change is `()`. I think adding method for the property getter isn't a good thing. Methods used for more complicated structures.

**Comment by amitzur** [comment]
@Artem-applitools any reason we are not merging this?

**Comment by Artem-applitools** [comment]
@amitzur I wasn't sure if it break the JS tests so haven't merged it yet.

**Comment by amitzur** [comment]
@Artem-applitools you can merge it, this is compatible in JS

**Comment by amitzur** [comment]
@Artem-applitools ping

**Pull request merged**